android/HelloPush/README.md

```diff
@@ -0,0 +1,166 @@
+
+1. Log into the UnifiedPush Server console.
+1. In the ```Applications``` view, click ```Create Application```.
+1. In the ```Name``` and ```Description``` fields, type values for the application and click ```Create```.
+1. When created, under the application click ```No variants```.
+1. Click ```Add Variant```.
+1. In the ```Name``` and ```Description``` fields, type values for the Android application variant.
+1. Click ```Android``` and in the ```Server Key``` and ```Sender ID``` fields type the values assigned to the project by FCM.
+1. Click ```Add```.
+1. When created, expand the variant name and make note of the ```Server URL```, ```Variant ID```, and ```Secret```.
+
+### Customize
+
+The project source code must be customized with the unique metadata assigned to the application variant by the AeroGear UnifiedPush Server and FCM.
+
+1. Download the ```google-services.json``` file from the Firebase Console and place it into the ```/app``` folder.
+1. Open ```/path/to/helloworld/android/app/src/main/assets/push-config.json``` for editing.
+1. Enter the application variant values allocated by the AeroGear UnifiedPush Server and FCM for the following constants:
+
+    ```js
+    {
+        "pushServerURL": "pushServerURL (e.g http(s)//host:port/context)",
+        "android": {
+            "senderID": "senderID (e.g Firebase Sender ID only for android)",
+            "variantID": "variantID (e.g. 1234456-234320)",
+            "variantSecret": "variantSecret (e.g. 1234456-234320)"
+        }
+    }
+    ```
+1. Save the file.
+
+### Running the app
+
+#### Prerequisites
+
+The AeroGear UnifiedPush Server must be running before the application is deployed to ensure that the device successfully registers with the AeroGear UnifiedPush Server on application deployment.
+
+#### Deploy for testing
+
+The application can be tested on physical Android devices only; push notifications are not available for Android emulators. To deploy, run and debug the application on an Android device attached to your system, on the command line enter the following:
+
+```shell
+$ cd /path/to/helloworld/android
+$ ./gradlew clean build
+$ ./gradlew installDebug
+```
+
+#### Send a Push Message
+
+You can send a push notification to your device using the AeroGear UnifiedPush Server console by completing the following steps:
+
+1. Log into the UnifiedPush Server console.
+2. Click ```Send Push```.
+3. From the ```Applications``` list, select the application.
+4. In the ```Messages``` field, type the text to be sent as the push notification.
+5. Click ```Send Push Notification```.
+
+![import](../cordova/doc/compose-message.png)
+
+## How does it work?
```

Comment:
I believe this section should be removed or changed to the official documentation we are writing, so that we won't have duplicates